### PR TITLE
Update setuptools to >=71.0.4

### PR DESCRIPTION
**Pull request by cdce8p**
## Description

Update setuptools version used for tests and build.
Fixes https://github.com/pylint-dev/pylint/security/dependabot/2

https://setuptools.pypa.io/en/latest/history.html#v71-0-4

### CODE REVIEW

File: requirements_test_min.txt [hunk 1]
```diff
@@ -9,4 +9,4 @@ pytest-timeout~=2.3
 towncrier~=23.11
 requests
 # Voluntary for test purpose, not actually used in prod, see #8904
-setuptools==41.6.0
+setuptools;python_version>='3.12'
```
Comment (jacobtylerwalls) [inline requirements_test_min.txt:12]: I'm in [favor](https://github.com/pylint-dev/pylint/pull/8904#discussion_r1284632741) of removing the version pin, but can I ask why limit the python version?

Comment (cdce8p) [reply requirements_test_min.txt:12]: Until `3.12`, `setuptools` was bundled with Python itself.

Comment (Pierre-Sassoulas) [reply requirements_test_min.txt:12]: This is the test dependencie set to this specific version as a regression test (see comment above) I think the threat reported is not applicable in this case.

Comment (jacobtylerwalls) [reply requirements_test_min.txt:12]: Got it, I must manually uninstalled it to reproduce that issue 😅

**Review by jacobtylerwalls — APPROVED** [review]

**Review by Pierre-Sassoulas — COMMENTED** [review]
Thank you Marc I did not realize that the pyproject.toml needed an upgrade (I only saw the test dependencie where it's not applicable imo).

**Review by jacobtylerwalls — APPROVED** [review]

**Pull request merged**